src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        point = toVector(point, "Could not convert 'point' to vector.")
+
+        pq = trimesh.proximity.ProximityQuery(self.mesh)
+
+        dist = abs(pq.signed_distance([point.coordinates])[0])
+
+        return dist
+
+    @property
+    def dimensionality(self):
+        return 2
+
+    @cached_property
+    def size(self):
+        return self.mesh.area
+
+    def getFlatOrientation(self, pos):
+        """ Get a flat orientation at a point in the region.
+
+        Given a point on the surface of the mesh, returns an orientation that aligns
+        an instance's z axis with the normal vector of the face containing that point.
+
+        If ``pos`` is not within ``self.tolerance`` of the surface of the mesh, a
+        ``RejectionException`` is raised.
+        """
+        prox_query = trimesh.proximity.ProximityQuery(self.mesh)
+
+        _, distance, triangle_id = prox_query.on_surface([pos.coordinates])
+
+        if distance > self.tolerance:
+            raise RejectionException("Attempted to get flat orientation of a mesh away from a surface.")
+
+        face_normal_vector = self.mesh.face_normals[triangle_id][0]
+
+        transform = trimesh.geometry.align_vectors([0,0,1], face_normal_vector)
+        orientation = tuple(scipy.spatial.transform.Rotation.from_matrix(transform[:3,:3]).as_euler('ZXY'))
+
+        return orientation
+
+    ## Utility Methods ##
+    @lru_cache(maxsize=None)
+    def getVolumeRegion(self):
+        """ Return a region equivalent to this one, except as a MeshVolumeRegion"""
+        return MeshVolumeRegion(self.mesh, self.name, orientation=self.orientation, \
+            tolerance=self.tolerance, centerMesh=False, onDirection=self.onDirection, engine=self.engine)
+
+    def getSurfaceRegion(self):
+        """ Returns this object, as it is already a MeshSurfaceRegion"""
+        return self
+
+class BoxRegion(MeshVolumeRegion):
+    """ Region in the shape of a rectangular cuboid, i.e. a box. 
+
+    By default the unit box centered at the origin and aligned with the axes is used.
+
+    Args:
+        name: An optional name to help with debugging.
+        dimensions: An optional 3-tuple, with the values representing width, length, height respectively.
+          The mesh will be scaled such that the bounding box for the mesh has these dimensions.
+        position: An optional position, which determines where the center of the region will be.
+        rotation: An optional Orientation object which determines the rotation of the object in space.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        tolerance: Tolerance for collision computations.
+        engine: Which engine to use for mesh operations. Either "blender" or "scad".
+    """
+    def __init__(self, name=None, dimensions=None, position=None, rotation=None, orientation=None,
+                 tolerance=1e-8, onDirection=None, engine="blender"):
```

Comment:
Might be better to use `**kwargs` so as not to keep this signature synchronized with `MeshVolumeRegion`. In that case the docstring can link to `MeshVolumeRegion` for the specification of the optional parameters.